sounds/views.py

```diff
@@ -125,10 +127,18 @@ def remixed(request):
 
 
 def random(request):
-    sound_id = Sound.objects.random()
-    if sound_id is None:
+    sounds = get_random_sounds_from_solr(1)
+    sound_obj = None
+    if sounds:
+        try:
+            sound_obj = Sound.objects.get(id=sounds[0]['id'])
+        except ObjectDoesNotExist:
+            pass
+    if sound_obj is None:
+        sound_id = Sound.objects.random()
+        sound_obj = Sound.objects.get(pk=sound_id)
+    if sound_obj is None:
```

Comment:
The line above is going to raise `Sound.DoesNotExist` if it doesn't work, not return `None`